server/controllers/ChallengeController.js

```diff
@@ -2,119 +2,94 @@ const db = require("../models/index");
-            return -1;
-    }
+  switch (difficulty) {
+    case "easy":
+      return 0;
+    case "medium":
+      return 1;
+    case "hard":
+      return 2;
+    default:
+      return -1;
+  }
 }
 
-
 exports.findAll = async (req, res) => {
-    try {
-        const challengesData = await Challenge.findAll();
-
-        const challenges = challengesData.map(challengeData => {
-            const challengeDescription = JSON.parse(challengeData.description);
-            return {
-                id: challengeData.id,
-                title: challengeData.title,
-                difficulty: diffToNum(challengeData.difficulty),
-                outcome: challengeDescription.outcome,
-                keyPatterns: challengeDescription.keyPatterns,
-                generalDescription: challengeDescription.generalDescription,
-                usageScenarios: challengeDescription.usageScenarios,
-                expectedFunctionality: challengeDescription.expectedFunctionality
-            }
-        });
-
-        res.status(200).json(challenges);
-    }
-    catch (error) {
-        res.status(500).json({ error: error.message });
-    }
+  const challengesData = await Challenge.findAll();
+  const challenges = challengesData.map((challengeData) => {
+    const challengeDescription = JSON.parse(challengeData.description);
+    return {
+      id: challengeData.id,
+      title: challengeData.title,
+      difficulty: diffToNum(challengeData.difficulty),
+      outcome: challengeDescription.outcome,
+      keyPatterns: challengeDescription.keyPatterns,
+      generalDescription: challengeDescription.generalDescription,
+      usageScenarios: challengeDescription.usageScenarios,
+      expectedFunctionality: challengeDescription.expectedFunctionality,
+    };
+  });
 
-}
+  res.status(200).json(challenges);
+};
 
 exports.findOne = async (req, res) => {
-	try {
-        const challenge_id = req.params.id;
-
-        const challengeData = await Challenge.findOne({
-            where: { 
-                id: challenge_id ,
-                // userId: req.user.id //Set up properly after done authentication
-            },
-        });
-
-        const challengeDescription = JSON.parse(challengeData.description);
-
-        // Convert the challenge into proper format
-        const challenge = {
-            id: challengeData.id,
-            title: challengeData.title,
-            difficulty: diffToNum(challengeData.difficulty),
-            outcome: challengeDescription.outcome,
-            keyPatterns: challengeDescription.keyPatterns,
-            generalDescription: challengeDescription.generalDescription,
-            usageScenarios: challengeDescription.usageScenarios,
-            expectedFunctionality: challengeDescription.expectedFunctionality
-        }
-
-        res.status(200).json(challenge);
-    }
-	catch (error) {
-        res.status(500).json({ error: error.message });
-    }
-}
+  const challenge_id = req.params.id;
```

Comment:
Same here, why remove the try-catch?

Reply:
Hey Alex, the try-catch blocks were really repetitive so I extracted them into a middleware to do the error handling. They are all the same code, so why repeat it for every route?